code/modules/antagonists/obsessed/obsessed.dm

```diff
@@ -0,0 +1,334 @@
+	else if(oldmind.assigned_role in GLOB.engineering_positions)
+		prefered_roles = GLOB.engineering_positions
+	else if(oldmind.assigned_role in GLOB.medical_positions)
+		prefered_roles = GLOB.medical_positions
+	else if(oldmind.assigned_role in GLOB.science_positions)
+		prefered_roles = GLOB.science_positions
+	else if(oldmind.assigned_role in GLOB.supply_positions)
+		prefered_roles = GLOB.supply_positions
+	//Gimmicks are special, there might be not enough of them to find coworker and civilians are technically their coworkers
+	else if((oldmind.assigned_role in GLOB.civilian_positions) || (oldmind.assigned_role in GLOB.gimmick_positions))
+		prefered_roles = GLOB.civilian_positions
+
+	for(var/mob/living/carbon/human/H in GLOB.player_list)
+		if(!SSjob.GetJob(H.mind.assigned_role) || H.mind == oldmind || length(H.mind.antag_datums))
+			continue //the jealousy target has to have a job, and not be the obsession or obsessed.
+		if(H.mind.assigned_role in prefered_roles)
+			prefered_coworkers += H
+			continue
+		other_coworkers += H.mind
+
+	var/mob/living/carbon/human/H
+	if(prefered_coworkers.len)//find someone in the same department
+		H = pick(prefered_coworkers)
+	else if(other_coworkers.len)//find someone who works on the station
+		H = pick(other_coworkers)
+
+	if(H)
+		target = H.mind
+		return
+
+/datum/objective/spendtime //spend some time around someone, handled by the obsessed trauma since that ticks
+	name = OBJECTIVE_STALK
+	var/timer = 5 MINUTES
+	var/target_name
+
+/datum/objective/spendtime/New(name)
+	. = ..()
+	target_name = name
+
+/datum/objective/spendtime/update_explanation_text()
+	if(timer == initial(timer))//just so admins can mess with it
+		timer += pick(-600, 0)
+	if(!owner.has_antag_datum(/datum/antagonist/obsessed))
+		qdel(src)
+		return
+	explanation_text = "Spend [DisplayTimeText(timer)] around [target_name] while they're alive."
+
+
+/datum/objective/spendtime/check_completion()
+	var/datum/antagonist/obsessed/O = owner.has_antag_datum(/datum/antagonist/obsessed)
+	return completed || (O?.total_time_stalking >= timer)
+
+/datum/objective/polaroid //take a picture of the target with you in it.
+	name = OBJECTIVE_PHOTOGRAPH
+
+/datum/objective/polaroid/update_explanation_text()
+	..()
+	if(target && target.current)
+		explanation_text = "Escape with a photo of [target], taken while they're alive."
+
+/datum/objective/polaroid/check_completion()
+	if(..())
+		return TRUE
+	var/list/all_items = owner.current.GetAllContents()	//this should get things in cheesewheels, books, etc.
+	for(var/obj/item/photo/P in all_items)
+		if(P.picture && (target.current in P.picture.mobs_seen) && !(target.current in P.picture.dead_seen))
+			return TRUE
+	return FALSE
+
+/datum/objective/heirloom_thief
+	name = OBJECTIVE_STEAL
+	var/datum/weakref/target_ref
+
+/datum/objective/heirloom_thief/New(obj/item/target)
+	target_ref = WEAKREF(target)	//item might be deleted at any point so it's unsafe to store strong refence
```

Comment:
I think this should store a strong reference and use a signal to detect deletion, this way you can also easily know if the item is unrecoverably gone immediately.

Reply:
I don't really care if the item is destroyed, I don't really want to track it.

Reply:
Make the target indestructible